src/taxonomy/api/hooks/api.js

```diff
@@ -1,20 +1,53 @@
 // @ts-check
-import { useQuery } from '@tanstack/react-query';
+import { useQuery, useMutation } from '@tanstack/react-query';
 import { camelCaseObject, getConfig } from '@edx/frontend-platform';
 import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { downloadDataAsFile } from '../../../utils';
 
 const getApiBaseUrl = () => getConfig().STUDIO_BASE_URL;
-const getTaxonomyListApiUrl = new URL('api/content_tagging/v1/taxonomies/?enabled=true', getApiBaseUrl()).href;
+const getTaxonomyListApiUrl = () => new URL('api/content_tagging/v1/taxonomies/?enabled=true', getApiBaseUrl()).href;
+const getExportTaxonomyApiUrl = (pk, format) => new URL(
+  `api/content_tagging/v1/taxonomies/${pk}/export/?output_format=${format}`,
+  getApiBaseUrl(),
+).href;
 
 /**
  * @returns {import("../types.mjs").UseQueryResult}
  */
-const useTaxonomyListData = () => (
+export const useTaxonomyListData = () => (
   useQuery({
     queryKey: ['taxonomyList'],
-    queryFn: () => getAuthenticatedHttpClient().get(getTaxonomyListApiUrl)
+    queryFn: () => getAuthenticatedHttpClient().get(getTaxonomyListApiUrl())
       .then(camelCaseObject),
   })
 );
 
-export default useTaxonomyListData;
+export const useExportTaxonomy = () => {
+  /**
+   * Calls the export request and downloads the file.
+   *
+   * Extra logic is needed to download the exported file,
+   * because it is not possible to download the file using the Content-Disposition header
+   * Ref: https://medium.com/@drevets/you-cant-prompt-a-file-download-with-the-content-disposition-header-using-axios-xhr-sorry-56577aa706d6
+   *
+   * @param {import("../types.mjs").ExportRequestParams} params
+   * @returns {Promise<void>}
+   */
+  const exportTaxonomy = async (params) => {
+    const { pk, format, name } = params;
+    const response = await getAuthenticatedHttpClient().get(getExportTaxonomyApiUrl(pk, format));
+    const contentType = response.headers['content-type'];
+    let fileExtension = '';
+    let data;
+    if (contentType === 'application/json') {
+      fileExtension = 'json';
+      data = JSON.stringify(response.data, null, 2);
+    } else {
+      fileExtension = 'csv';
+      data = response.data;
+    }
+    downloadDataAsFile(data, contentType, `${name}.${fileExtension}`);
```

Comment:
Calling the endpoint with the parameter `download=1` you could get the filename and extension from the backend using the `Content-Disposition` and simplify a bit here.

Another alternative is to leave the front end as it is and remove the `download` option to simplify the backend.

What do you think?

Reply:
@rpenido We need to call the endpoint using Axios because the call needs to be authenticated. I have tried using `download=1`, but doesn't works, [here is the explanation](https://medium.com/@drevets/you-cant-prompt-a-file-download-with-the-content-disposition-header-using-axios-xhr-sorry-56577aa706d6). I've added that [as a comment](https://github.com/open-craft/frontend-app-course-authoring/pull/4/files/2070743175408ae9a844fa575b024ab88cec3605#diff-51e2c0d3363c83c5d167493ee852ee13f16564ad2c70189c74d66f8adaca8cd6R29).

> Another alternative is to leave the front end as it is and remove the download option to simplify the backend.

I think yes, we need to remove that because it will no longer be necessary, but it would be done in a follow-up task, I'm already running out of time here CC @bradenmacdonald

Reply:
I also thought that the authentication would be a problem, but @bradenmacdonald is right. Making a request directly to the backend using the authenticated browser section worked (in my setup).

Reply:
@rpenido How did you make the request? I have tried `window.location.href` way but doesn't work. CC @bradenmacdonald 


https://github.com/open-craft/frontend-app-course-authoring/assets/11827305/89eef993-e6ac-4ec4-bedc-5b971fb63622

Reply:
@ChrisChV Weird. It works for me - I just went to http://studio.local.overhang.io:8001/api/content_tagging/v1/taxonomies/1/export/?output_format=json&download=1 (the tutor devstack equivalent URL) in my browser and it downloaded a file.

If you go to http://localhost:18010 are you logged in to Studio?

Reply:
@ChrisChV  I think you are using an old version of openedx-learning, and you're missing this commit: https://github.com/openedx/openedx-learning/commit/d66d6294c1b27fbb285cda365802925f528bfe39

Reply:
> If you go to http://localhost:18010/ are you logged in to Studio?

Yes, that is it. Now it works, thanks! That simplifies things